### ISSUE TITLE
Clumsy resource.New options

### ISSUE THREAD

**Issue by MrAlias** [open-telemetry/opentelemetry-go#1690]
The [`WithFromEnv`](https://pkg.go.dev/go.opentelemetry.io/otel/sdk@v0.18.0/resource#WithFromEnv), [`WithHost`](https://pkg.go.dev/go.opentelemetry.io/otel/sdk@v0.18.0/resource#WithHost), and [`WithTelemetrySDK`](https://pkg.go.dev/go.opentelemetry.io/otel/sdk@v0.18.0/resource#WithTelemetrySDK) options that are passed to [`New`](https://pkg.go.dev/go.opentelemetry.io/otel/sdk@v0.18.0/resource#New) all accept a `Detector` as an argument. This `Detector` is stated to be used as an alternate to the builtin `Detector` that finds the related resource attributes, or if nil will omit those resource attributes. This feels very clumsy.

For instance, there is not guarantee that a `Detector` is passed to `WithTelemetrySDK` that will detect `telemetry.sdk.*` attributes. It could detect whatever attributes it wants. The only meaningful effect including this option will have is that it will not include the builtin detector (nil or otherwise).

Based on this behavior, it seems like these options should be changed to [`Without*` versions](https://github.com/open-telemetry/opentelemetry-go/blob/main/CONTRIBUTING.md#options) of themselves. If a user wants to provide a custom detector to override these attributes they can use the [`WithDetectors`](https://pkg.go.dev/go.opentelemetry.io/otel/sdk@v0.18.0/resource#WithDetectors) option.

### Proposal

- Replace [`WithFromEnv`](https://pkg.go.dev/go.opentelemetry.io/otel/sdk@v0.18.0/resource#WithFromEnv) with
   ```
   // WithoutFromEnv disables the builtin FromEnv detector from being run.
   func WithoutFromEnv() Option { /* ... */ }
   ```
   (though this might be removed entirely in #1689)
- Replace [`WithHost`](https://pkg.go.dev/go.opentelemetry.io/otel/sdk@v0.18.0/resource#WithHost) with
   ```
   // WithoutHost disables the builtin Host detector from being run.
   func WithoutHost() Option { /* ... */ }
   ```
- Replace [`WithTelemetrySDK`](https://pkg.go.dev/go.opentelemetry.io/otel/sdk@v0.18.0/resource#WithTelemetrySDK) with
   ```
   // WithoutTelemetrySDK disables the builtin TelemetrySDK detector from being run.
   func WithoutTelemetrySDK() Option { /* ... */ }
   ```

**Comment by seh** [comment]
I think this approach won't age well, as we will have users relying on these functions that would later turn out to have no effect if we were to revise which things the default detectors handle automatically. In other words, say that we later decide that we shouldn't detect the host-related details automatically. We would have users calling `WithoutHost`, which would no longer have any effect, and we'd probably then need to introduce `withHost` again.

**Comment by MrAlias** [comment]
@seh what is your proposal?

**Comment by MadVikingGod** [comment]
What if we remove the three `WithFromEnv`, `WithHost`, and `WithTelemetrySDK` and use the default ones unless `WithDetectors` is used.  

To make it convenient to add your own detector and have the builtins, we could have an exported slice of BuiltinDetectors.  The New call would then look like this: 
```go
resource.New(
    WithDetectors(BuiltinDetectors...),
    WithDetectors(MyCustomDetector),
)
```

**Comment by MadVikingGod** [comment]
@seh I put in a PR just toying around with this.  Can you take a look and see if that format would be more approachable?

**Comment by MrAlias** [comment]
Closed with #1810
